client/src/main/java/io/pravega/client/stream/impl/StreamCut.java

```diff
@@ -29,5 +31,14 @@
     private final Stream stream;
     @Getter(value = AccessLevel.PACKAGE)
     private final Map<Segment, Long> positions;
-    
+
+    @VisibleForTesting
+    public boolean validate(List<String> segmentNames) {
+        boolean containsAll = true;
+        for(Segment s: positions.keySet()) {
+            containsAll = containsAll && segmentNames.contains(s.getScopedName());
```

Comment:
Are we expecting `segmentNames` to be a large list? This is a O(N) search.

Maybe we can try one or more of the following:
- Drop `containsAll` and inside the loop, if `!segmentNames.contains(...)` then return false. Return true at the end of the method. This guarantees a quick exit as soon as we find a non-compliant one.
- Make/Force SegmentNames to be a `Set` (or `HashSet`)?